### ISSUE TITLE
expr: 50n \> -51 is producing different result than GNU expr

### ISSUE THREAD

**Issue by cakebaker** [uutils/coreutils#5378]
GNU expr:
```
$ expr 50n \> 50
1
$ expr 50n \> 51
0
$ expr 50n \> -50
1
$ expr 50n \> -51
0
```
uutils expr:
```
$ cargo run expr 50n \> 50
1
$ cargo run expr 50n \> 51
0
$ cargo run expr 50n \> -50
1
$ cargo run expr 50n \> -51
1
```

**Comment by Luv-Ray** [comment]
I have tested the behavior of different versions of coreutils:

GNU coreutils 9.1-1:
```shell
$ expr b \> -b
1
$ expr b \> -bb
1
$ expr 50n \> -51
1
```
GNU coreutils 9.1-12, 9.3, 9.4:
```shell
$ expr b \> -b
1
$ expr b \> -bb
0
$ expr 50n \> -51
0
```
I don't know if it's intentional or just a bug, I can't find a reason to specifically deal with `-`.

**Comment by tertsdiepraam** [comment]
Where does 9.1-12 come from? Is that from GNU? I'm trying to find some release notes about it but I can't find it.

**Comment by Luv-Ray** [comment]
9.1-12 from [RPM](https://rpmfind.net/linux/RPM/fedora/updates/38/x86_64/Packages/c/coreutils-9.1-12.fc38.x86_64.html)

**Comment by tertsdiepraam** [comment]
Hmm, those do indeed seem to have some additional patches. That makes it a bit hard to track. Do you have any way to test with the GNU versions without Fedora patches? Maybe via `nix` for example? In a week I can try this myself again.

**Comment by Luv-Ray** [comment]
I encountered some issues while trying to compile coreutils directly from GNU, and I am currently attempting to fix them. Is the goal to find the first version which producing `0` for `expr b \> -bb`?

**Comment by tertsdiepraam** [comment]
Yeah, I'm curious about the patch that introduced the change. We might also be able to find it through release notes or something like that. Thanks for all the effort!

**Comment by Luv-Ray** [comment]
I found that there haven't been significant changes to `expr` within the 9.1 version range, and the reason of the different results may caused by **local**. See [my question](https://stackoverflow.com/questions/77307041/why-does-gnu-expr-produce-0-for-expr-a-aa) in stack overflow.

The result is same as GNU in `LC_COLLATE=C`.
```
$ LC_COLLATE=C ./target/debug/expr a \> -a
1
$ LC_COLLATE=C ./target/debug/expr a \> -aa
1
```

**Comment by tertsdiepraam** [comment]
Ohh of course! Great find! I guess we should add it to this then: https://github.com/uutils/coreutils/issues/3997

**Comment by cakebaker** [comment]
Fixed with https://github.com/uutils/coreutils/pull/8292
